Jellyfin.Server/Program.cs

```diff
@@ -163,25 +163,30 @@ private static ServerApplicationPaths CreateApplicationPaths(StartupOptions opti
             // ELSE IF windows, use <%APPDATA%>/jellyfin
             // ELSE IF $XDG_DATA_HOME then use $XDG_DATA_HOME/jellyfin
             // ELSE    use $HOME/.local/share/jellyfin
-            if (options.DataDir != null)
+            if (!string.IsNullOrEmpty(dataDir = options.DataDir))
```

Comment:
TBH I would rather we won't try to be so clever here... even if compilers would be really bad at optimizing stuff - this happens only once at startup and at most gives a few milliseconds of extra time. But it hurts reading a lot.

Seems like I'm disagreeing on @Bond-009's thoughts here, though :)

Reply:
I wouldn't have done it inline, but I stand by my point not to call GetEnvironmentVariable twice.

Reply:
I initially thought these GetEnvironmentVariable() calls would be replaced by the compiler with pointers into the process env block..  but as the env names are unknown until runtime that isn't possible and a function call must be made for each reference.

I prefer this approach tbh, but happy to change (again)..  Just let me know :)

Reply:
Now !inline.